client/src/components/client/Registration.js

```diff
@@ -1,87 +1,204 @@
   const licenseType = useRef();
   const licenseDate = useRef();
+
+  const sendForm = () => {
+    if (validateForm()) {
+      var userName = lastName.current.value + " " + firstName.current.value;
+      var userEmail = email.current.value;
+      var userPassword = pwd.current.value;
+      var userLicenseType = licenseType.current.value;
+      var userLicenseDate = new Date(licenseDate.current.value);
+      var user = {
+        userName,
+        userEmail,
+        userPassword,
+        userLicenseDate,
+        userLicenseType,
+      };
+      axios
+        .post("http://localhost:3004/auth/register", user)
+        .then(function (response) {
+          console.log(response);
+        })
+        .catch((error) => {
+          console.error("There was an error!", error);
+        });
+      console.log(user);
+      handleClose();
+    } else console.log("not valid data");
+  };
+
+  const validateForm = () => {
+    if (
+      validateName(firstName.current.value) &&
+      validateName(lastName.current.value) &&
+      validateEmail(email.current.value) &&
+      validatePassword(pwd.current.value) &&
+      validateLicenseType(licenseType.current.value) &&
+      validateLicenseDate(licenseDate.current.value)
+    )
+      return true;
+    else return false;
+  };
+  const validateName = (name) => {
+    var template = /([A-Za-zÉÁŰÚŐÓÜÖÍéáűúőóüöí]{2,30}$)/g;
+    if (template.test(name)) {
+      return true;
+    } else return false;
+  };
+  const validateEmail = (email) => {
+    const re = /^(([^<>()\]\\.,;:\s@"]+(\.[^<>()\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(email));
+  };
+  const validatePassword = (password) => {
+    if (password.length >= 5) return true;
+    else return false;
+  };
+  const validateLicenseType = (type) => {
+    if (type === "-") return false;
+    else return true;
+  };
+  const validateLicenseDate = (date) => {
+    var licenseDates = date.split("-");
+    var nowDates = getCurrentDate().split("-");
+    var licenseYear = licenseDates[0];
+    var licenseMonth = licenseDates[1];
+    var licenseDay = licenseDates[2];
+    var nowYear = nowDates[0];
+    var nowMonth = nowDates[1];
+    var nowDay = nowDates[2];
+    if (
+      licenseYear <= nowYear &&
+      licenseMonth <= nowMonth &&
+      licenseDay <= nowDay
+    )
+      return true;
+    else return false;
+  };
+
+  function getCurrentDate(separator = "-") {
+    let newDate = new Date();
+    let date = newDate.getDate();
+    let month = newDate.getMonth() + 1;
+    let year = newDate.getFullYear();
+
+    var now = `${year}${separator}${
+      month < 10 ? `0${month}` : `${month}`
+    }${separator}${date}`;
+    return now;
+  }
+
   return (
-    <Modal.Dialog>
-      <Modal.Header closeButton>
-        <Modal.Title>Regisztráció</Modal.Title>
-      </Modal.Header>
+    <>
+      <Nav.Item variant="primary">
```

Comment:
Csak akkor kell megjeleniteni a gombot, ha meg nem lepett be, hasonloan a loginhoz.